### DISCUSSION TITLE
Deprecating service credentials

### DISCUSSION THREAD

**Discussion by dremsol** [mindsphere/mindconnect-nodejs#296] [Q&A] [answered]
Hi all,

I have been using service credentials for years for both our productive tenant, as our development tenant.  As service credentials seem to be deprecated I was wondering how to switch to application credentials such that I can use the CLI for our productive tenant. The examples show how to create an app for the development tenant in order to create the app credentials but how would this work for our productive tenant? Does this mean I need to deploy the application to our operator tenant and provision it to our productive environment?

**Comment by sn0wcat** [comment, marked as answer]
You can provision a dummy "CLI" app , enable app credentials and use that in the CLI, yes :) 

![image](https://user-images.githubusercontent.com/1426617/134920396-1838f6fb-cee9-4cae-b1cf-1c064aa9145a.png)

**Comment by dremsol** [comment]
I'm getting an error with app credentials created in dev tenant. I followed following [example](https://opensource.mindsphere.io/docs/mindconnect-nodejs/cli/setting-up-the-cli.html#tab1anchor3) with following result

![image](https://user-images.githubusercontent.com/34475288/134960722-9a310a12-1073-4a7b-8d0e-c4a0c61c8eac.png)

We do have early access on SDS in our dev environment. Could it be that additional settings are necessary?

**Reply by sn0wcat** [reply]
you probably have a typo somewhere, perhaps in the version number?

**Reply by sn0wcat** [reply]
here is an example of a whole bunch of working credentials, the 3rd one from above is e.g. app credential from operations tenant.

![image](https://user-images.githubusercontent.com/1426617/134980692-fed32b56-567b-4bcf-a731-e49c493a53bd.png)

**Reply by dremsol** [reply]
I indeed screwed up with the version number. Works like a charm now for all 3 tenants :) Thanks for the support
